api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationSearchAdviserTest.java

```diff
@@ -0,0 +1,446 @@
+        authenticateUserWithLocationProperty();
+        assertAuthenticatedNormalUser();
+
+        List<String> result = getUuidsOfLocations(locationService.getAllLocations());
+        assertEquals(1, result.size());
+        assertEquals(location3Uuid, result.get(0));
+    }
+
+    @Test
+    public void getAllLocations_nonAdminWithoutLocationPropertyShouldGetOnlySessionLocation() {
+        authenticateUserWithoutLocationProperty();
+        assertAuthenticatedNormalUser();
+
+        setSessionLocationID(4);
+
+        List<String> result1 = getUuidsOfLocations(locationService.getAllLocations());
+        assertEquals(1, result1.size());
+        assertEquals(location4Uuid, result1.get(0));
+
+        setSessionLocationID(2);
+
+        List<String> result2 = getUuidsOfLocations(locationService.getAllLocations());
+        assertEquals(1, result2.size());
+        assertEquals(location2Uuid, result2.get(0));
+    }
+
+    @Test
+    public void getLocations_adminShouldGetAllLocations() {
+        authenticateAdmin();
+        assertAuthenticatedAdmin();
+
+        List<String> result1 = getUuidsOfLocations(locationService.getLocations(searchQuery));
+        assertEquals(3, result1.size());
+        assertBothCollectionsAreEqual(unretiredLocationUuids, result1);
+
+        List<String> result2 = getUuidsOfLocations(locationService.getLocations(searchQuery, null, null, true, null, null));
+        assertEquals(5, result2.size());
+        assertBothCollectionsAreEqual(allLocationUuids, result2);
+    }
+
+    @Test
+    public void getLocations_nonAdminWithLocationPropertyShouldGetOnlyAccessLocation() {
+        authenticateUserWithLocationProperty();
+        assertAuthenticatedNormalUser();
+
+        List<String> result1 = getUuidsOfLocations(locationService.getLocations(searchQuery));
+        assertEquals(1, result1.size());
+        assertEquals(location3Uuid, result1.get(0));
+
+        List<String> result2 = getUuidsOfLocations(locationService.getLocations(searchQuery, null, null, true, null, null));
+        assertEquals(1, result2.size());
+        assertEquals(location3Uuid, result1.get(0));
+    }
+
+    @Test
+    public void getLocations_nonAdminWithoutLocationPropertyShouldGetOnlySessionLocation() {
+        authenticateUserWithoutLocationProperty();
+        assertAuthenticatedNormalUser();
+
+        setSessionLocationID(2);
+
+        List<String> result1 = getUuidsOfLocations(locationService.getLocations(searchQuery));
+        assertEquals(1, result1.size());
+        assertEquals(location2Uuid, result1.get(0));
+        List<String> result2 = getUuidsOfLocations(locationService.getLocations(searchQuery, null, null, true, null, null));
+        assertEquals(1, result2.size());
+        assertEquals(location2Uuid, result1.get(0));
+
+        setSessionLocationID(4);
+
+        List<String> result3 = getUuidsOfLocations(locationService.getLocations(searchQuery));
+        assertEquals(0, result3.size());
```

Comment:
Why you are not getting the session location here? but in the next line, you are getting session location.

Reply:
Because it is retired. That's why I test two variations of `getLocations` method. Should I put it in a separate test, which mentions that the result is filtered out because session location is retired?